src/main/java/com/google/cloud/tools/appengine/operations/AppYamlProjectStaging.java

```diff
@@ -74,15 +74,30 @@ public void stageArchive(AppYamlProjectStageConfiguration config) throws AppEngi
       String runtime = findRuntime(config);
       if ("flex".equals(env)) {
         stageFlexibleArchive(config, runtime);
-      } else if ("java11".equals(runtime)) {
-        stageStandardArchive(config);
-      } else {
-        // I don't know how to deploy this
+        return;
+      }
+      if ("java11".equals(runtime)) {
+        boolean isJar = config.getArtifact().getFileName().toString().endsWith(".jar");
+        if (isJar) {
+          stageStandardArchive(config);
+          return;
+        }
+        if (hasCustomEntrypoint(config)) {
+          stageStandardBinary(config);
+          return;
+        }
+        // I cannot deploy non-jars without custom entrypoints
         throw new AppEngineException(
-            "Cannot process application with runtime: "
-                + runtime
-                + (Strings.isNullOrEmpty(env) ? "" : " and env: " + env));
+            "Cannot process application with runtime: java11,"
+                + " non-jar artifact: "
+                + config.getArtifact().toString()
+                + " must have a custom entrypoint defined in app.yaml");
```

Comment:
```suggestion
            "Cannot process application with runtime: java11. Non-jar artifact "
                + config.getArtifact().toString()
                + " must have a custom entrypoint defined in app.yaml");
```
And I find that it's desirable to put file path at the end of the message if possible, because file path can often be super long.

Reply:
rearranged this so file name is at end.